core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.spec.ts

```diff
@@ -0,0 +1,306 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for contribution and review backend api service.
+ */
+
+import { HttpClientTestingModule, HttpTestingController }
+  from '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+
+describe('Contribution and review backend API service', () => {
+  let carbas: ContributionAndReviewBackendApiService;
+  let http: HttpTestingController;
+
+  let suggestion1 = {
+    suggestion_id: 'suggestion_id_1',
+    target_id: 'skill_id_1',
+  };
+  let opportunityDict1 = {
+    skill_id: 'skill_id_1',
+    skill_description: 'skill_description_1',
+  };
+  let testSuggestionsBackendObject = {
+    suggestions: [
+      suggestion1
+    ],
+    target_id_to_opportunity_dict: {
+      skill_id_1: opportunityDict1,
+    },
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    carbas = TestBed.inject(ContributionAndReviewBackendApiService);
+    http = TestBed.inject(HttpTestingController);
+  });
+
+  afterEach(() => {
+    http.verify();
+  });
+
+  describe('fetching suggestions from the backend', () => {
+    let successHandler: jasmine.Spy<jasmine.Func>;
+    let failureHandler: jasmine.Spy<jasmine.Func>;
+
+    beforeEach(() => {
+      successHandler = jasmine.createSpy('success');
+      failureHandler = jasmine.createSpy('failure');
+    });
+
+    it('should fetch submitted question suggestions', fakeAsync(() => {
+      spyOn(carbas, 'fetchSubmittedSuggestionsAsync').and.callThrough();
+      const url = '/getsubmittedsuggestions/skill/add_question';
+
+      carbas.fetchSuggestionsAsync(
+        'SUBMITTED_QUESTION_SUGGESTIONS'
+      ).then(successHandler, failureHandler);
+
+      const req = http.expectOne(url);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(testSuggestionsBackendObject);
+
+      flushMicrotasks();
```

Comment:
Nit: These lines are all associated with the test "action", so consider grouping them together. Here and below.

Reply:
done